08-generic_cache/03_ttl/cache_internal_test.go

```diff
@@ -0,0 +1,47 @@
+package cache
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCacheRead(t *testing.T) {
+	// Initialise the cache and prepare data.
+	cache := New[int, string]()
+	cache.data[5] = &entry[string]{value: "fünf", expires: time.Now().Add(time.Minute)}
+
+	// Testing key is present.
+	got, err := cache.Read(5, func(i int) (string, error) {
+		return cache.data[5].value, nil
```

Comment:
```suggestion
		return "lima", nil
```

Make it explicit that, since the key is there, we're not using the callback